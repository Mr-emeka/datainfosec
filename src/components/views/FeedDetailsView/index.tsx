import React from 'react'
import MainLayout from '../../layout/mainLayout'
import { NewsProps } from '../../../types'
import Comments from '../../custom/comments'

export interface IProps {
    news: NewsProps
}
const FeedDetailsView: React.FC<IProps> = ({ news }) => {
    return <MainLayout title={news?.title} description={news?.description}>
        <div className="feedDetails">
            <section >
                {news.image_url && <img alt="news_image" src={news?.image_url} className="feedDetails__img" />}
                <h1 className="feedDetails__title mb-3">{news?.title}</h1>
                <div className="mb-2">
                    <p>{news?.description}</p>
                </div>
                <hr />
                <section className="comment">
                    <h4>Comments</h4>
                    <Comments newsId={news.uuid} />
                </section>
            </section>
        </div>
    </MainLayout>
}
export default FeedDetailsView;