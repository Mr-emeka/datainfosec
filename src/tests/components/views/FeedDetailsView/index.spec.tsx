import React from "react";
import { shallow } from "enzyme";
import FeedDetailsView from "../../../../components/views/FeedDetailsView";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("Feed Details View", () => {
  it("should render correctly", () => {
    const Component = shallow(<FeedDetailsView news={{}} />);
    expect(Component).toMatchSnapshot();
  });
});
