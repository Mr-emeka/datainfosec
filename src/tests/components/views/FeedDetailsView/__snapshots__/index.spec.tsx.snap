// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed Details View should render correctly 1`] = `ShallowWrapper {}`;
